### PR TITLE
Fix authors page in ISBN form

### PR DESCRIPTION
**Pull request by mvillivald**
1. Changed the appearance of the Authors page. The page is now divided into two blocks. One block is for form and buttons, and the other is for displaying the data in the list of cards.
2. Now the form is more intuitive, the user can more clearly understand what roles have been added and which one belongs to a particular author.
3. Several roles can be added to a single person.
4. Implemented the option to delete roles one by one without deleting a whole person.
5. Added clarifying comments to ListComponent.js.

### CODE REVIEW

**Review by aatuny — CHANGES_REQUESTED** [review]

File: src/frontend/components/ListComponent.js [hunk 1]
```diff
@@ -643,3 +639,11 @@ export default function (props) {
...
 	}
 }
+
+const mapStateToProps = state => {
+	return {
+		values: getFormValues('isbnIsmnRegForm')(state)
+	};
+};
+
+export default connect(mapStateToProps)(ListComponent);
```
[the remark this answers is not in the document]
Comment (villivald) [reply src/frontend/components/ListComponent.js]: Done, connect is removed for now

File: src/frontend/components/ListComponent.js [hunk 2]
```diff
@@ -184,47 +191,30 @@ export default function (props) {
...
-										<span> No Authors Added</span>
-									</Grid>
-								)}
-						</div>
+						(values && values.authors && values.authors.length) ? values.authors.map(item => (
+							<Card key={item.id} className={classes.authorCard}>
+								<CardContent className={classes.authorCardContent}>
+									<div className={classes.authorCardBlock}>
+										<Typography className={classes.authorCardLabel}>
+											Nimi:
```
[the remark this answers is not in the document]
Comment (villivald) [reply src/frontend/components/ListComponent.js]: Done

File: src/frontend/components/form/constants.js [hunk 3]
```diff
@@ -43,6 +43,14 @@ export const FORMATS = {
 	BOTH: 'both'
 };
 
+export const ROLES = [
+	{label: <FormattedMessage id="publicationRegistration.form.authors.role.authors"/>, value: 'tekijä'},
```
[the remark this answers is not in the document]
Comment (villivald) [reply src/frontend/components/form/constants.js]: Done, I deleted the unneeded constant array

File: src/frontend/components/form/render/renderContactDetail.js [hunk 4]
```diff
@@ -126,51 +127,85 @@ export default connect(state => ({
...
 					fields.push(author);
 					clearFields(undefined, false, false, 'authorGivenName', 'authorFamilyName', 'role');
 				}
 			}
 		}
 	};
 
+	const handleDeleteAuthor = (author, role) => {
+		if (author.role.length === 1) {
+			setErrors();
```
[the remark this answers is not in the document]
Comment (villivald) [reply src/frontend/components/form/render/renderContactDetail.js]: Done

File: src/frontend/components/form/render/renderContactDetail.js [hunk 5]
```diff
@@ -220,29 +255,42 @@ export default connect(state => ({
...
-								return (
-									<Chip
-										key={`${item.authorGivenName}${index}`}// eslint-disable-line react/no-array-index-key
-										label={`${item.authorGivenName} ${item.authorFamilyName}`}
-										onDelete={() => fields.remove(index)}
-									/>
-								);
-							})}
+						<div className={classes.authorsContainer}>
+							<h2>Tekijät</h2>
```
Comment (aatuny) [inline src/frontend/components/form/render/renderContactDetail.js:259]: Use FormattedMessage to apply multilanguage support.

File: src/frontend/components/form/render/renderContactDetail.js [hunk 6]
```diff
@@ -220,29 +255,42 @@ export default connect(state => ({
...
+							<p className={classes.zeroAuthorsInfo}>
+								{values && values.authors && values.authors.length ? null : 'Tekijöitä ei ole lisätty vielä, lisää tekijä täyttämällä tiedot ja painamalla "Lisää" nappia'}
```
Comment (aatuny) [inline src/frontend/components/form/render/renderContactDetail.js:261]: Apply FormattedMessage to here also.

File: src/frontend/components/subComponents/cards/AuthorCard.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	const classes = useStyles();
+	const values = value.value;
+
+	return (
+		(values && values.length) ? values.map(author => (
+			<Card key={author.id} className={classes.authorCard}>
+				<CardContent className={classes.authorCardContent}>
+					<div className={classes.authorCardBlock}>
+						<Typography className={classes.authorCardLabel}>
+							Nimi:
```
Comment (aatuny) [inline src/frontend/components/subComponents/cards/AuthorCard.js:20]: Apply FormattedMessage.

File: src/frontend/components/subComponents/cards/AuthorCard.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+							<Chip
+								key={uuidv4()}
+								label={<FormattedMessage id={`publicationRegistration.form.authors.role.${role}`}/>}
+								color="default"
+							/>
+						))}
+					</div>
+				</CardContent>
+			</Card>
+		)) : 'Tekijöitä ei ole lisätty'
```
Comment (aatuny) [inline src/frontend/components/subComponents/cards/AuthorCard.js:38]: Apply FormattedMessage

File: src/frontend/components/subComponents/cards/AuthorCard.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+					<div className={classes.authorCardBlock}>
+						<Typography className={classes.authorCardLabel}>
+							Nimi:
+						</Typography>
+						<Typography>
+							{author.givenName} {author.familyName}
+						</Typography>
+					</div>
+					<div className={classes.authorCardBlock}>
+						<Typography className={classes.authorCardLabel}>Roolit:</Typography>
```
Comment (aatuny) [inline src/frontend/components/subComponents/cards/AuthorCard.js:27]: Apply FormattedMessage

**Review by aatuny — CHANGES_REQUESTED** [review]

**Review by aatuny — APPROVED** [review]

**Pull request merged**